plugins/infinispan93/src/main/java/org/radargun/service/Infinispan93HotRodTransactional.java

```diff
@@ -0,0 +1,89 @@
+package org.radargun.service;
+
+import javax.transaction.TransactionManager;
+
+import org.radargun.logging.Log;
+import org.radargun.logging.LogFactory;
+import org.radargun.traits.Transactional;
+
+/**
+ * @author Diego Lovison &lt;dlovison@redhat.com&gt;
+ */
+public class Infinispan93HotRodTransactional implements Transactional {
+   protected static final Log log = LogFactory.getLog(Infinispan93HotRodTransactional.class);
+   protected static final boolean trace = log.isTraceEnabled();
+
+   protected final Infinispan93HotrodService service;
+
+   public Infinispan93HotRodTransactional(Infinispan93HotrodService service) {
+      this.service = service;
+   }
+
+   @Override
+   public Configuration getConfiguration(String cacheName) {
+      return service.isCacheTransactional(cacheName) ?
+         Configuration.TRANSACTIONAL : Configuration.NON_TRANSACTIONAL;
+   }
+
+   @Override
+   public Transaction getTransaction() {
+      return new Infinispan93HotRodTransactional.Tx();
+   }
+
+   protected <T> TransactionManager getTransactionManager(T resource) {
+      if (resource instanceof HotRodOperations.HotRodCache) {
+         return ((HotRodOperations.HotRodCache) resource).noReturn.getTransactionManager();
+      } else {
+         throw new IllegalArgumentException("Invalid resource: " + String.valueOf(resource));
```

Comment:
@alanfx we should use `resource instanceof HotRodOperations.HotRodCache` to avoid a ClassCastException.
In this case, should we return null?

Reply:
@diegolovison Can you make this happen or are you just worried it might happen? I'm trying to understand how the wrong type of cache would be passed into the HotRod service.

Reply:
I consider the cast check as a good practice and prefer an error instead of null.
Removed.

Reply:
I feel like if this error occurs, we have bigger problems!

Reply:
In this case, it means that the dev implemented something wrong. IMO, it is ok keep as is because `ClassCastException` or `IllegalArgumentException` will stop the code execution.